modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/resources/META-INF/resources/hooks/useSyncValue.es.js

```diff
@@ -36,9 +36,10 @@ export const useSyncValue = (newValue, isDelay = true, forceValue) => {
 					value !== newValue &&
 					previousValueRef.current !== newValue
 				) {
-					previousValueRef.current = newValue;
 					setValue(newValue);
 				}
+
+				previousValueRef.current = newValue;
```

Comment:
Hey @markocikos, we used this when we were gradually migrating from Metal.js to React.js, as the Field component can be used with two data streams, one where it just receive the value and one where it sends a new one and receives it again, historically, had some bugs like typing quickly caused loss of values, that's because it had Metal.js at the top and React.js at the bottom, so you couldn't trust when Metal.js would send the new data, because it was too slow , since we migrated everything to React.js today, maybe we don't even need `useSyncValue` anymore, we would have to test to see if these behaviors still exist.

Changing the behavior of `previousValueRef.current` would also have to check if it is safe, maybe it is, I don't remember well, but it would be good to test this with other fields.

Reply:
@matuzalemsteles I see. I tried to replace `useSyncValue` with `useState` just in `RichText`, and it seems to work. This seems like a much safer change to make, `useSyncValue` is used in 10 or so components. Do you know what I need to test to confirm all is ok?

Reply:
I force-pushed changes:
- I reverted the previous value handling change on `useSyncValue`
- in `RichText` we are using `useState` instead of `useSyncValue`, https://github.com/liferay-data-engine/liferay-portal/pull/541/commits/e3b1d5e2cb3fa14fb589649b4a3785b6ec15e549

I retested steps of this bug and all seems to work correctly.

Reply:
Hmm, yeah I think it's something they should try to get rid of now that things are more stable but in that case, maybe we would have to make the component controlled, stateless in the component just getting the `value` values and using `onChange` to trigger a change, because if we keep an internal `useState` it won't work for the case when you're creating a Form Builder and adding a field and editing the field's predefined value, in that case when there's a predefined value it updates the content of the preview field that is in the builder.

I would try to follow these steps to test if it's working fine as well as quickly typing in the predefined value to see if it's not losing value or slowing down.

Reply:
The problem is that we are maintaining a [hidden `<input>`](https://github.com/liferay/liferay-portal/blob/master/modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/resources/META-INF/resources/RichText/RichText.es.js#L97-L102) with current value, I'm guessing this is key for form submission. If we are making this controlled, this would need to be moved to parent component. I suggested going the opposite path in [this comment](https://issues.liferay.com/browse/LPS-128735?focusedCommentId=2464529&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-2464529), to remove `value` prop and keep it in internal state. But I do see your point with preview, it's a bit tricky to ideally set concerns :thinking: 

In any case, this is beyond the scope of this PR :slightly_smiling_face:

Reply:
Yeah, this is complicated. I think the team will have to deal with this now.

Reply:
> because if we keep an internal useState it won't work for the case when you're creating a Form Builder and adding a field and editing the field's predefined value, in that case when there's a predefined value it updates the content of the preview field that is in the builder.

Yep, you are right, this case breaks simple `useState`. It also breaks the original solution, with `previousValueRef.current` behavior change.

I'll try a bit more to find a solution.